src/Table/Table.js

```diff
@@ -0,0 +1,71 @@
+import { instanceOfComponent } from '@dhis2/prop-types'
+import React from 'react'
+import css from 'styled-jsx/css'
+import propTypes from 'prop-types'
+
+import { Consumer } from './tableContext'
```

Comment:
I would like us to move towards a more standards compliant way to import modules.

Only certain bundlers allow:

1. Omitting the suffix (`.js` in this case)
2. Automatic resolution of `index.js` files in directories

E.g. **Webpack** (ES6+NodeJS supported) supports it. **Rollup** (Only ES6+ supported) does not.

The reason for that is that this is **NodeJS Module Resolution** semantics. It is strictly speaking not valid for ECMAScript Modules.

I started digging in to this when I realised that Rollup did not accept the way our `import` statements are designed in **ui-core** since we do both **1.** and **2.** above. 🙅‍♂ 

The reason our ES6 bundle (`build/es`) works at all is because Babel saves our asses by fixing all our `import` statements to:

```js
import { Consumer } from './tableContext.js';
```

I don't think we should be mixing NodeJS semantics in pure ECMAScript libraries. It is confusing and increases our reliance on our tooling to save us from ourselves.

### Further reading

[1] https://nodejs.org/dist/latest-v10.x/docs/api/modules.html
[2] https://www.ecma-international.org/ecma-262/6.0/#sec-imports
[3] https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/import#Syntax

Reply:
We don't have to do this in this PR. We can do it for the entire library in one go and merge that in. We should keep it backwards compatible and at the same time think about re-organising the code from `src/${Component}/index.js` to e.g. `src/${Component}.js` + `src/${Component}/styles.js` (e.g.).

Reply:
> We should keep it backwards compatible and at the same time think about re-organising the code from src/${Component}/index.js to e.g. src/${Component}.js + src/${Component}/styles.js (e.g.).

Yeah, this is what I wanted to do!

> The reason for that is that this is NodeJS Module Resolution semantics. It is strictly speaking not valid for ECMAScript Modules.

So in the future we should append the `.js` suffix, did I get that right?

Reply:
> Yeah, this is what I wanted to do!

I remember. 👍

> So in the future we should append the .js suffix, did I get that right?

Yes.